streams/src/main/java/org/apache/kafka/streams/kstream/Transformer.java

```diff
@@ -17,48 +17,96 @@
+     * {@link ProcessorContext#schedule(long) schedule itself} for periodical calls (cf. {@link #punctuate(long)}), and
+     * to access attached {@link StateStore}s.
+     * <p>
+     * Note, that {@link ProcessorContext} is updated in the background with the current record's meta data.
+     * Thus, it only contains valid record meta data when accessed within {@link #transform(Object, Object)}.
      *
-     * @param context the context; may not be null
+     * @param context the context
      */
     void init(final ProcessorContext context);
 
     /**
      * Transform the record with the given key and value.
+     * Additionally, any {@link StateStore state} that is {@link KStream#transform(TransformerSupplier, String...)
+     * attached} to this operator can be accessed and modified
+     * arbitrarily (cf. {@link ProcessorContext#getStateStore(String)}).
+     * <p>
+     * If more than one output record should be forwarded downstream {@link ProcessorContext#forward(Object, Object)},
+     * {@link ProcessorContext#forward(Object, Object, int)}, and
+     * {@link ProcessorContext#forward(Object, Object, String)} can be used.
+     * If not record should be forwarded downstream, {@code transform} can return {@code null}.
      *
      * @param key the key for the record
      * @param value the value for the record
-     * @return new value; if null no key-value pair will be forwarded to down stream
+     * @return new {@link KeyValue} pair&mdash;if {@code null} no key-value pair will
+     * be forwarded to down stream
      */
     R transform(final K key, final V value);
 
     /**
-     * Perform any periodic operations and possibly generate a key, if this processor {@link ProcessorContext#schedule(long) schedules itself} with the context
-     * during {@link #init(ProcessorContext) initialization}.
+     * Perform any periodic operations and possibly generate new {@link KeyValue} pairs if this processor
+     * {@link ProcessorContext#schedule(long) schedules itself} with the context during
+     * {@link #init(ProcessorContext) initialization}.
+     * <p>
+     * To generate new {@link KeyValue} pairs {@link ProcessorContext#forward(Object, Object)},
+     * {@link ProcessorContext#forward(Object, Object, int)}, and
+     * {@link ProcessorContext#forward(Object, Object, String)} can be used.
+     * <p>
+     * Note that {@code punctuate} is called based on <it>stream time</it> (i.e., time progresses with regard to
+     * timestamps return by the used {@link TimestampExtractor})
+     * and not based on wall-clock time.
      *
-     * @param timestamp the stream time when this method is being called
-     * @return new value; if null it will not be forwarded to down stream
+     * @param timestamp the stream time when {@code punctuate} is being called
+     * @return must return {@code null}&mdash;otherwise, a {@link StreamsException exception} will be thrown
```

Comment:
Thanks for reporting this @mihbor (I guess this was a mix up with `ValueTransformer`). Can you open an PR to fix it?